index.d.ts

```diff
@@ -16,20 +18,20 @@ export type Mapper<ValueType, KeyType, MappedValueType> = (
 export default function pProps<
 	KeyType extends unknown,
 	ValueType extends unknown,
-	MappedValueType = ValueType
+	MappedValueType = PromiseResult<ValueType>
 >(
-	input: Map<KeyType, PromiseLike<ValueType> | ValueType>,
+	input: Map<KeyType, ValueType>,
 	mapper?: Mapper<ValueType, KeyType, MappedValueType>,
 	options?: Options
 ): Promise<Map<KeyType, MappedValueType>>;
 export default function pProps<
-	KeyType extends string,
-	ValueType extends unknown,
-	MappedValueType = ValueType
+	InputType extends object,
+	ValueType extends InputType[keyof InputType],
+	MappedValueType = PromiseResult<ValueType>
 >(
-	input: {[key in KeyType]: PromiseLike<ValueType> | ValueType},
-	mapper?: Mapper<ValueType, KeyType, MappedValueType>,
+	input: InputType,
+	mapper?: Mapper<ValueType, keyof InputType, MappedValueType>,
```

Comment:
I think that it was a mistake that `Mapper` can receive `Promise` values in `element` parameter. This isn't actually possible, `element` is always the resolved value of the corresponding `Promise` from `input`.

Reply:
That's what I have expected as well, but it actually receives unresolved value (see #10).